src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private static final double DEFAULT_SIZE_GROW = 1.5;
+
+    private int size;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[MAX_ITEMS_NUMBER];
+    }
+
+    private T[] growSize() {
+        int newCapacity = 0;
+        int oldCapacity = values.length;
+        newCapacity = (int) (oldCapacity * DEFAULT_SIZE_GROW);
+
+        T[] newValues = (T[]) new Object[newCapacity];
+        System.arraycopy(values, 0, newValues, 0, size);
+        return newValues;
+    }
+
     @Override
     public void add(T value) {
+        if (size == values.length) {
+            values = growSize();
```

Comment:
You have problem with naming here. Method growSize() is expected to be void or return int according to it's name but it returns array.
Please change naming to be more obvious (like getGrownArray() etc.) or rework it to be void so you will use like
if (size == values.length) {
             growArraySize();